config/src/test/scala/com/typesafe/config/impl/ConfigSubstitutionTest.scala

```diff
@@ -924,7 +924,7 @@ class ConfigSubstitutionTest extends TestUtils {
 
     @Test
     def substSelfReferenceIndirect() {
-        val obj = parseObject("""a=1, b=${a}, a=${b}""")
+        val obj = parseObject("""b=${a}, a=${b}""")
```

Comment:
I've been trying to figure this out and the current behavior really doesn't make sense to me.

You're saying this is supposed to throw an exception because there's a cycle?

    a=1, b=${a}, a=${b}

And this is supposed to resolve to `a=2`?

    a=1, a=2

The reason that doesn't make sense to me because in the first case we're saying you can't assume these will be resolved left to right and in the second case we're saying they will be. It's really inconsistent and I don't think that should be able to be the case. If `a=2` in the second example then it's just as easily the case that `a=1, b=${a}, a=${b}` can be resolved left to right such that a=1 and then b=a is resolved to 1 and then a=b can be resolved to 1 since we've already resolved b as equal to 1.  Or if you can't resolve left to right like that then you have no idea in the other example whether a will be resolved to 1 or to 2. You either need to always assume ordering or never.